src/Statistics.jl

```diff
@@ -675,11 +703,18 @@ corm(x::AbstractVecOrMat, xmean, y::AbstractVecOrMat, ymean, vardim::Int=1) =
 
 # cor
 """
-    cor(x::AbstractVector)
+    cor(itr::Any)
 
 Return the number one.
 """
-cor(x::AbstractVector) = one(real(eltype(x)))
+function cor(itr::Any)
+    if Base.IteratorEltype(itr) isa Base.HasEltype && isconcrete(eltype(itr))
+        return one(real(eltype(itr)))
+    else
+        return one(real(eltype(collect(itr))))
+    end
+end
+cor(x::AbstractVector{T}) where {T} = one(real(T))
```

Comment:
No need to change this:
```suggestion
cor(x::AbstractVector) = one(real(eltype(x)))
```